### PR TITLE
Remove deprecate warning for Upsampling

### LINKED ISSUE #11452: Upsampling modules deprecated
All Upsampling modules in torch/nn/modules/upsampling.py are listed as deprecated with the following message "nn.Upsampling is deprecated. Use nn.functional.interpolate instead.". This means there is currently no way to include upsampling in a Sequential model without getting a deprecation warning. For this reason, I think the Upsample module should not be deprecated. However, it could be renamed to Interpolate to be consistent with the changes in the functional API.

### PR DESCRIPTION
**Pull request by ailzhang**
Fixes #11452 .

Based on the discussion with @SsnL  and @soumith , we want to bring back Upsample as a module instead of introducing a new nn.interpolate module for now. If anyone want to do downsample, they should use `nn.functional.interpolate ` instead.

### CODE REVIEW

**Review by ssnl — COMMENTED** [review]
That was fast! Could you add a `.. note::` to the doc and tell users if they want to resize/downsample, they should use `torch.nn.functional.interpolate`? (Btw, does `interpolate` support general resizing, i.e., to arbitrary size?)

File: torch/nn/modules/upsampling.py [hunk 1]
```diff
@@ -194,10 +186,6 @@ class UpsamplingBilinear2d(Upsample):
         size (tuple, optional): a tuple of ints `(H_out, W_out)` output sizes
         scale_factor (int, optional): the multiplier for the image height or width
 
-    .. warning::
```
Comment (SsnL) [inline torch/nn/modules/upsampling.py]: I think you missed something here. `UpsamplingNearest2d` is deprecated already before your change. So we should keep this.

**Comment by SsnL** [comment]
@ailzhang Okay. But we can still link to F.interpolate :)

File: torch/nn/modules/upsampling.py (hunk 1 above)
Comment (ailzhang) [reply torch/nn/modules/upsampling.py]: Oh thanks, will add it back now. :D

**Comment by vadimkantorov** [comment]
@SsnL A crazy idea: how about a generic `nn.Wrap(F.interpolate, scale_factor = 2)` - can make sense for stateless function, will unlbock people on absence of some modules, and even crazier `nn.Sequential().add(F.interpolate, **kwargs)` can exist.

File: torch/nn/modules/upsampling.py [hunk 2]
```diff
@@ -50,8 +50,8 @@ class Upsample(Module):
         0.3.1. Since then, the default behavior is ``align_corners = False``.
         See below for concrete examples on how this affects the outputs.
 
-    .. warning::
-        This class is deprecated in favor of :func:`~nn.functional.interpolate`.
+    .. note::
+        If you want down sampling, you should use :func:`~nn.functional.interpolate`.
```
Comment (SsnL) [inline torch/nn/modules/upsampling.py]: nit: downsampling is a single word.

**Review by ssnl — APPROVED** [review]

**Comment by nehz** [comment]
Any updates on this ?

**Comment by gchanan** [comment]
@nehz this has already been committed.

**Comment by nehz** [comment]
I'm still seeing it on the master and 1.0.1 branch ?
`https://github.com/pytorch/pytorch/blob/master/torch/nn/modules/upsampling.py`
` warnings.warn("nn.{} is deprecated. Use nn.functional.interpolate instead.".format(self.name))`

**Comment by SsnL** [comment]
blaming https://github.com/pytorch/pytorch/pull/14123 cc @eellison

**Comment by gchanan** [comment]
@ailzhang ?

**Comment by ailzhang** [comment]
Yea this got added back accidentally in another PR. I'm sending a quick patch to fix. Thanks @SsnL @gchanan !

**Pull request closed without merging**